src/build-dep-graph.js

```diff
@@ -75,6 +75,7 @@ function secondPass(workspaceMeta) {
 
 async function buildDepGraph({
   workspaceCwd,
+  shouldPruneDeps = true,
```

Comment:
what does it mean to prune deps? based on the added test (which sets the option to false) , I'm not sure

Reply:
before this new feature, monorepo-next was only concerned with packages in the monorepo, and excluded everything else. This brings in the ability to deal with non-monorepo packages for defragging.